src/Umbraco.Core/Persistence/Repositories/RecycleBinRepository.cs

```diff
@@ -63,6 +63,7 @@ public virtual bool EmptyRecycleBin()
                 FormatDeleteStatement("cmsContentVersion", "ContentId"),
                 FormatDeleteStatement("cmsContentXml", "nodeId"),
                 FormatDeleteStatement("cmsContent", "nodeId"),
+                //TODO: Why is this being done? We just delete this exact data in the next line

```

Comment:
I had a suspicion that this had to do with some constraints causing problems when deleting nodes that were marked as `trashed` but hadn't actually had their `parentId` set correctly.

Looking up the change it seems like that could be the case:
https://github.com/umbraco/Umbraco-CMS/commit/30e12c1367357ac05bb7b265b0f4d8e71ed3b47d

Not sure whether this applies anymore though, but leaving it in there for now.